### DISCUSSION TITLE
C++ linking error: ... AudioFFTNotesAudio ... first defined here

### DISCUSSION THREAD

**Discussion by MauiKano** [pschatzmann/arduino-audio-tools#1114] [Q&A] [answered]
I have a C++ program with multiple  .h  and .cpp . In one header file I use the audio and fft classes. In the corresponding header file I include :

#pragma once
#include "PluginManager.h"
#include "AudioTools.h"
#include "AudioLibs/AudioRealFFT.h" // or AudioKissFFT

It compiles but at the linking stage I get an error 
........platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld: .pio/build/mhetesp32minikit-ota/src/plugins/FFTPlugin.cpp.o:(.data._ZN11audio_tools13AudioFFTNotesE+0x0): multiple definition of `audio_tools::AudioFFTNotes'; .pio/build/mhetesp32minikit-ota/src/main.cpp.o:(.data._ZN11audio_tools13AudioFFTNotesE+0x0): first defined here

I understand what this means but I do not see how I can resolve that even when  I guess its a pretty simple mistake I make.
Appreciate a concrete hint.

**Comment by pschatzmann** [comment, marked as answer]
I did not test all the optional classes to support multiple compilation units, after all an Arduino sketch consists usually of just one single sketch file

So I suggest that you clone the project and after doing and testing the corrections, submit a pull request, so that others can profit from it as well.

Correcting this is easy: just change
MusicalNotes AudioFFTNotes;
to
static MusicalNotes AudioFFTNotes;
in AudioFFT.h

**Comment by MauiKano** [comment]
That resolved the linker conflict. I'm able now to complile and link without error. Thanks for the quick and constructive reply. I did make a pull request as proposed. Hopefully correctly, I'm not experienced in GitHub.
